lib/beaker/subcommand.rb

```diff
@@ -6,15 +6,41 @@ module Beaker
   class Subcommand < Thor
     SubcommandUtil = Beaker::Subcommands::SubcommandUtil
 
+    def initialize(*args)
+      super
+      @@config = SubcommandUtil.load_config()
+    end
+
     desc "init", "Initialises the beaker test environment configuration"
     option :hypervisor, :type => :string, :required => true
     long_desc <<-LONGDESC
-      Initialises a beaker environment configuration
+    Initialises a beaker environment configuration
     LONGDESC
     def init()
       SubcommandUtil.verify_init_args(options)
       SubcommandUtil.require_tasks()
       SubcommandUtil.init_hypervisor(options)
+      SubcommandUtil.store_config(options)
+      SubcommandUtil.delete_config([:provisioned, :hosts])
+    end
+
+    desc "provision", "Provisions the beaker test environment"
+    option :force, :type => :boolean
+    option :validate, :type => :boolean, :default => true
+    option :configure, :type => :boolean, :default => true
+    long_desc <<-LONGDESC
+    Provisions a beaker environment
+    LONGDESC
+    def provision()
+      provisioned = @@config.transaction { @@config[:provisioned] }
```

Comment:
The transaction stuff here is because I'm using YAMLStore to persist data across runs. I know that in other parts of beaker, we just use simple file reading/writing, but I was curious what your guys' thoughts are on using YAMLStore. If you'd rather I just use simple file reading/writing, that's totally fine 👍